utils/wasm-gen/src/generator/syscalls/invocator.rs

```diff
@@ -529,52 +515,139 @@ impl<'a, 'b> SysCallsInvocator<'a, 'b> {
         ]
     }
 
-    fn insert_sys_call_instructions(&mut self, instructions: &[Instruction]) -> Result<()> {
+    fn insert_sys_calls(&mut self) -> Result<()> {
         log::trace!(
-            "Random data before inserting sys-call's invoke instructions - {}",
+            "Random data before inserting sys-calls invoke instructions - {}",
             self.unstructured.len()
         );
 
-        let last_funcs_idx = self.module.count_code_funcs() - 1;
-        let mut insert_into_func_no = self.unstructured.int_in_range(0..=last_funcs_idx)?;
+        let code_funcs = self.module.count_code_funcs();
+        let insert_into_funcs: Vec<_> = (0..code_funcs)
+            .filter(|idx| !self.call_indexes.is_custom_func(*idx))
+            .collect();
 
-        // Do not insert into custom newly generated function, but only into pre-defined
-        // internal functions.
-        //
-        // This loop will definitely end, because there are only 4 custom functions (3 for gear entry points
-        // and one for precise reservation send) and minimal amount of internal functions is 15.
-        while self.call_indexes.is_custom_func(insert_into_func_no) {
-            insert_into_func_no = self.unstructured.int_in_range(0..=last_funcs_idx)?;
+        let syscalls_to_insert = self
+            .sys_call_imports
+            .clone()
+            .into_iter()
+            .map(|(syscall, (amount, _))| {
+                iter::repeat(syscall)
+                    .take(amount as usize)
+                    .collect::<Vec<_>>()
+            })
+            .flatten();
```

Comment:
use `.flat_map()`, clippy should show how
```
$ cargo clippy -p gear-wasm-gen
warning: called `map(..).flatten()` on `Iterator`
```

Reply:
maybe also enable CI in .github/workflows/PR.yml?
like here: https://github.com/gear-tech/gear/pull/3262/commits/f4f9965ac12c74e6bf93e1eb1ff13158491fded5